standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/dbinstall/AbstractDatabase.java

```diff
@@ -0,0 +1,302 @@
+package org.apache.hadoop.hive.metastore.dbinstall;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.hadoop.hive.metastore.tools.schematool.MetastoreSchemaTool;
+import org.junit.rules.ExternalResource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import sqlline.SqlLine;
+
+import java.io.BufferedReader;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.PrintStream;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+public abstract class AbstractDatabase extends ExternalResource {
```

Comment:
Do we want to keep this dependent to `ExternalResource` and junit APis in general?